python/ngen_init_config/src/ngen/init_config/deserializer.py

```diff
@@ -105,3 +107,21 @@ def from_json(cls, p: Path) -> Self:
     @classmethod
     def from_json_str(cls, s: str) -> Self:
         return cls.parse_raw(s)
+
+
+class GenericDeserializer(Base, ABC):
+    """
+    Stub for deserializing from a generic format.
+    Subclasses must implement the `from_file` and `from_str` abstract class methods.
+    See `ngen.init_config.core.Base` and `pydantic`'s documentation for configuration options.
+    """
+
+    @classmethod
+    @abstractmethod
+    def from_file(cls, p: Path, *_) -> Self:
```

Comment:
Note for reviewer, `*_` effectively means that subclasses that add arguments must make them keyword only / optional. The goal is improve uniformity of implementations.

Reply:
I would add that note in the extended section of the doc string.

Reply:
I'm not too worried. I probably would literally have written it as "Note to self:" because I would forget in the future haha.